pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -222,6 +239,36 @@ func (f *framework) RunPrefilterPlugins(
 	return nil
 }
 
+// RunScorePlugins runs the set of configured scoring plugins. It returns a map that
+// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// It also returns *Status, which is set to non-success if any of the plugins returns
+// a non-success status.
+func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.Node) (PluginToNodeScoreMap, *Status) {
+	pluginToNodeScoreMap := make(PluginToNodeScoreMap, len(f.scorePlugins))
+	for _, pl := range f.scorePlugins {
+		errCh := make(chan string, len(nodes))
+		plScores := make(NodeScoreList, len(nodes))
+		workqueue.ParallelizeUntil(context.TODO(), 16, len(nodes), func(index int) {
+			score, status := pl.Score(pc, pod, nodes[index].Name)
```

Comment:
IMO this could cause a potential problem, `pl` variable is a pointer and could be captured by the `ParallelizedUntil` function, you could run the code below in https://play.golang.org/p/Jc271gRPfdy

```go
func main() {
	values := []struct {
		i int
	}{
		{
			i: 1,
		},
		{
			i: 2,
		},
	}

	wg := sync.WaitGroup{}
	wg.Add(2)
	for _, value := range values {
		go func() {
			defer wg.Done()
			fmt.Println(value.i)
		}()
	}
	wg.Wait()
}

$ go run prog.go
./prog.go:25:16: loop variable value captured by func literal
Go vet exited.

2
2
```

We could solve this problem with the following pattern:

```go
for i := range f.scorePlugins {
  pl := f.scorePlugins[i]
  // ...
}
```

FYI: [Using goroutines on loop iterator variables](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables)